src/convert/traits.rs

```diff
@@ -249,3 +250,24 @@ impl<T: WasmAbi> WasmRet<T> {
         T::join(self.prim1, self.prim2, self.prim3, self.prim4)
     }
 }
+
+/// `TryFromJsValue` is a trait for converting a JavaScript value (`JsValue`)
+/// into a Rust type. It is particularly useful when dealing with JavaScript
+/// objects in WebAssembly, allowing for custom conversion logic that might
+/// be needed when interacting with JavaScript APIs or passing data between
+/// Rust and JavaScript.
+///
+/// This trait is intended as a solution to the limitations imposed by the
+/// automatic derivation of `TryFrom<JsValue>` in `wasm-bindgen`, which
+/// restricts the ability to implement custom conversions for ABI-bound structs.
+///
+/// Types implementing this trait must specify their conversion logic from
+/// `JsValue` to the Rust type, handling any potential errors that may occur
+/// during the conversion process.
```

Comment:
```suggestion
/// [`TryFromJsValue`] is a trait for converting a JavaScript value ([`JsValue`])
/// into a Rust type. It is used by the [`wasm_bindgen`](wasm_bindgen_macro::wasm_bindgen)
/// proc-macro to allow conversion to user types.
///
/// Types implementing this trait must specify their conversion logic from
/// [`JsValue`] to the Rust type, handling any potential errors that may occur
/// during the conversion process.
```

I removed the historical part, which I would like to pretend just never happened instead of having to remember to remove it at some later point when it becomes only confusing instead of useful for new users.

Also adjusted some of the wording implying that this is anything else then internal stuff for now.